client/src/components/RequestLunch.js

```diff
@@ -0,0 +1,276 @@
+					aria-labelledby="lunch-radio-buttons-group-label"
+					name="lunch-radio-buttons-group"
+					value={lunchOption}
+					onChange={handleChangeInputRadioButtons}
+				>
+					<Box sx={{ display: "flex" }}>
+						<FormControlLabel value="yes" control={<Radio />} label="Yes" />
+						<FormControlLabel value="no" control={<Radio />} label="No" />
+					</Box>
+				</RadioGroup>
+
+				{lunchOption == "yes" ? (
+					<>
+						{/* checkboxes for selecting any dietary restrictions */}
+						<FormLabel
+							id="restrictions-radio-buttons-group-label"
+							component="legend"
+						>
+							Do you have any dietary restrictions?
+						</FormLabel>
+						<Box sx={{ display: "flex" }}>
+							<FormGroup>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.milk}
+											onChange={handleChangeInputRestrictions}
+											name="milk"
+										/>
+									}
+									label="Milk"
+								/>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.eggs}
+											onChange={handleChangeInputRestrictions}
+											name="eggs"
+										/>
+									}
+									label="Eggs"
+								/>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.peanuts}
+											onChange={handleChangeInputRestrictions}
+											name="peanuts"
+										/>
+									}
+									label="Peanuts"
+								/>
+							</FormGroup>
+							<FormGroup>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.fish}
+											onChange={handleChangeInputRestrictions}
+											name="fish"
+										/>
+									}
+									label="Fish"
+								/>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.soy}
+											onChange={handleChangeInputRestrictions}
+											name="soy"
+										/>
+									}
+									label="Soy"
+								/>
+								<FormControlLabel
+									control={
+										<Checkbox
+											checked={dietaryRestrictions.wheat}
+											onChange={handleChangeInputRestrictions}
+											name="wheat"
+										/>
+									}
+									label="Wheat"
+								/>
+							</FormGroup>
+						</Box>
+
+						{/* text field so the user can input any others dietary restrictions */}
+						<TextField
+							id="dietary-restrictions"
+							name="dietary-restrictions"
+							label="Others"
+							margin="dense"
+							value={otherDietaryRestrictions}
+							onChange={handleChangeTextField}
+							autoComplete="off"
+						/>
+
+						{/* checkboxes for selecting any dietary requirements */}
+						<FormLabel
+							id="requirements-radio-buttons-group-label"
+							component="legend"
+						>
+							Do you have any dietary requirements?
+						</FormLabel>
+						<Box sx={{ display: "flex" }}>
+							<FormGroup>
+								<FormControlLabel
```

Comment:
**The Same in here from line 213 up-to line 256 ,,,,,,,, needs to be changed.**

Reply:
I used exactly the same approach as in the previous one  to solve this...